timelock-server/src/suiteTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -484,6 +491,37 @@ public void stressTestForPaxosEndpoints() {
         }
     }
 
+    @Test
+    public void leaderCanProcessMultiClientLeaderTimeRequest() {
+        TestableTimelockServer leader = cluster.currentLeaderFor(client.namespace());
+        MultiClientConjureTimelockService multiClientConjureTimelockService = leader.multiClientService();
+        Set<String> expectedNamespaces = ImmutableSet.of("alpha", "beta");
+        List<NamespacedLeaderTime> leaderTimes =
+                multiClientConjureTimelockService.leaderTimes(AUTH_HEADER, expectedNamespaces);
+        Set<String> namespaces =
+                leaderTimes.stream().map(NamespacedLeaderTime::getNamespace).collect(Collectors.toSet());
+        assertThat(namespaces).hasSameElementsAs(expectedNamespaces);
+
+        Set<LeadershipId> leadershipIds = leaderTimes.stream()
+                .map(NamespacedLeaderTime::getLeaderTime)
+                .map(LeaderTime::id)
+                .collect(Collectors.toSet());
+        assertThat(leadershipIds).hasSameSizeAs(expectedNamespaces);
+    }
+
+    @Test
+    public void leaderCanProcessMultiClientGetCommitTimestampRequest() {
+        MultiClientConjureTimelockService service =
+                cluster.currentLeaderFor(client.namespace()).multiClientService();
+        Set<String> expectedNamespaces = ImmutableSet.of("alpha", "beta");
+        List<NamespacedGetCommitTimestampsResponse> commitTimestamps =
+                service.getCommitTimestamps(AUTH_HEADER, getGetCommitTimestampsRequests(expectedNamespaces));
+        Set<String> namespaces = commitTimestamps.stream()
+                .map(NamespacedGetCommitTimestampsResponse::getNamespace)
+                .collect(Collectors.toSet());
+        assertThat(namespaces).hasSameElementsAs(expectedNamespaces);
```

Comment:
This is perhaps being a bit pedantic, but get commit timestamps have a leader ID hidden inside (it's the `logId` inside `LockWatchStateUpdate`). Perhaps worth a similar check to the one above?

Reply:
Done